webapp/src/components/App/Singlestudy/explore/Modelization/Areas/Hydro/utils.ts

```diff
@@ -156,3 +165,31 @@ function generateColumns(columnSuffix = ""): string[] {
   }
   return columns;
 }
+
+/**
+ * A custom hook that returns area options filtered by the given fields.
+ *
+ * @param fields - An array of objects containing areaId and coefficient.
```

Comment:
If you want to use JSDoc...

```
 * @param {Object[]} fields ...
 * @param {string} fields[].id ...
 * @param {number} fields[].coefficient ...
```

But in my option, in TS there is no necessity to describe arguments